### DISCUSSION TITLE
guard against hyperlog copying

### DISCUSSION THREAD

**Discussion by hackergrrl** [digidem/mapeo-desktop#540] [Ideas]
*Bad things* happen when a hyperlog's leveldb directory gets copied around.
@mafintosh had some advice for this on `#digidem`:

> to guard against the hyperlogs being copies bug, i usually save a hash in the
> level with a hash of the username + abs path. if the hash doesn't match the
> prev one i'll throw an error telling the user something is wrong

**Comment by gmaclennan** [comment]
One (clunky) solution: keep a separate store on the client that keeps track of the latest `seq` or other value in the hyperlog. If it differs, we know the hyperlog has been swapped out without our knowledge. People could still copy the entire app data folder between machines though, e.g. restoring from a backup onto a new machine...

any way to pick this up during replication?

**Comment by okdistribute** [comment]
Will this be obsolete with kappa-core/hypercore?

**Comment by hackergrrl** [comment]
I think this is still a danger, since your private key is stored in the multifeed directory, and its presence is how multifeed/hypercore knows who the "writer" is.
